src/mame/drivers/photoplysx.cpp

```diff
@@ -0,0 +1,103 @@
+// license:BSD-3-Clause
+// copyright-holders:
+/*
+	Funworld Photo Play Spirit Xtreme
+
+	CPU: Intel CELERON 2 GHz / 128 / 400 (SL6VR)
+	RAM: 256MB DDR 333
+	PCB: Intel 865G + Intel Extreme Graphics 2 + Intel FW82801FR
+	I/O: Cypress AN2131SC (12.000 MHz xtal) + Winbond W83627HF + Realtek RTL8101L
+	BIOS: 03/11/2009-I865G-6A79AD4EC-00 (Pm49FL004T-33JC)
+	Dongle: USB
+	Sound: C-Media CMI9761A
+*/
+
+#include "emu.h"
+#include "cpu/i386/i386.h"
+#include "screen.h"
+
+class photoplaysx_state : public driver_device

```

Comment:
Single-file drivers like this should keep stuff besides the `GAME`/`COMP`/etc. macros in an anonymous namespace.  It makes it harder for accidental name conflicts to happen, it allows clang to use more aggressive optimisations, and it speeds up linking.  (The improvement in link time is small per class, but it adds up eventually when you have as many sources as MAME has.)

Reply:
Can you please provide an example driver?
Thanks a lot!!

Reply:
There are no example drivers. It's never been a requirement to put single drivers in an anonymous namespace, and it won't be one going forward unless it is discussed with the rest of the MAME board. This is an arbitrary requirements that Vas has pulled out of nowhere.

Reply:
Most of the chess computer drivers are starting to go that way:
* https://github.com/mamedev/mame/blob/mame0225/src/mame/drivers/fidel_csc.cpp
* https://github.com/mamedev/mame/blob/mame0225/src/mame/drivers/saitek_cp2000.cpp
* https://github.com/mamedev/mame/blob/mame0225/src/mame/drivers/mephisto_academy.cpp

It’s not a hard requirement at this point, just a nice-to-do thing as it helps avoid annoying-to-diagnose class name conflicts (they don’t show up at all in single-driver builds and show up as blowing up on validation or start rather than a compile or link error in full builds), and it slightly improves build time.

Reply:
Fixed with PR #7355